spec/spec_helper.rb

```diff
@@ -21,6 +24,32 @@
 $LOAD_PATH << '../app/domain'
 
 RSpec.configure do |config|
+  config.before(:all) do
+    # enable FIPS mode
+    OpenSSL.fips_mode = true
+
+    # override ActiveSupport hash_digest_class with FIPS complaint method
+    ActiveSupport::Digest.hash_digest_class = OpenSSL::Digest::SHA1.new
+
+    # override Sprockets digest_class with FIPS complaint method
+    Sprockets::DigestUtils.module_eval do
+      def digest_class
+        OpenSSL::Digest::SHA256
+      end
+    end
+    Sprockets.config = Sprockets.config.merge(
+        digest_class: OpenSSL::Digest::SHA256
+    ).freeze
+
+    # override OpenIDConnect cache_key with FIPS complaint method
+    OpenIDConnect::Discovery::Provider::Config::Resource.module_eval do
+      def cache_key
+        sha256 = Digest::SHA256.hexdigest host
+        "swd:resource:opneid-conf:#{sha256}"
+      end
+    end
+  end
+
```

Comment:
Much of this code is repeated from above.  My comments from above would apply here too.  And I'd also wonder if we are missing some abstraction or better design that would allow us to avoid this repetition. 

Also, it's not clear why rspec needs this configuration.  Our application is what needs to be FIPS compliant.  Something has gone wrong somewhere, I would think, if this is requiring changes to our test code too.

Reply:
No need for abstraction.
appears that during development and rspec failure the fips changes were added to spec_helper. But it is taken from fips.rb.
So remove this piece of useless code.